### ISSUE TITLE
`/^\s*\*.*$/m` doesn't match whole line

### ISSUE THREAD

**Issue by zuckerruebe** [lark-parser/lark#1130]
`%ignore /^\s*\*.*$/m` doesn't match `<Space><Space>** Dependencies with other typez\n` but gives

![image](https://user-images.githubusercontent.com/11159050/161082292-aad4548b-54f6-4157-b143-7d4ed4e6929c.png)

It should match the whole line as a comment, though:

https://pythex.org/?regex=%5E%5Cs*%5C*.*%24&test_string=%20%20**%20Dependencies%20with%20other%20typez&ignorecase=0&multiline=1&dotall=0&verbose=0

**Comment by zuckerruebe** [comment]
The work-around I had to resort to was `/^\s*\*[^\n]*\n?/m`. It's not clear to me, however, why this version works and the previous one not.

**Comment by zuckerruebe** [comment]
Also, if is it correct to say that the `/m` flag only works with the Earley parser? If I remember correctly I had no luck at all with the LALR(1) parser. But I couldn't find anything in the docs to that effect. Cheers! <3

**Comment by erezsh** [comment]
No, it should work with both parsers similarly.

**Comment by zuckerruebe** [comment]
> It might help if you give the token a higher priority.

Intersting...aren't comments removed in a first, separate pass?

**Comment by erezsh** [comment]
No, that's true for LALR, but too simplistic for Earley. It keeps track of all the possible derivations, which means the same character can be ignored in one, and not ignored in another.

**Comment by erezsh** [comment]
Small note: It will be true for Earley if you use `lexer="basic"`.

**Comment by zuckerruebe** [comment]
> It might help if you give the token a higher priority.

I don't think it's possible to give weights to terminals when using Earley, or is it? The `%ignore` _does_ expect a terminal, doesn't it?
